examples/pdm_monorepo/MODULE.bazel

```diff
@@ -0,0 +1,40 @@
+module(
+    name = "example_pdm_monorepo",
+    version = "0.0.0",
+    compatibility_level = 1,
+)
+
+bazel_dep(name = "aspect_bazel_lib", version = "1.38.1")
+bazel_dep(name = "rules_python", version = "0.29.0")
+
+python = use_extension("@rules_python//python/extensions:python.bzl", "python")
+python.toolchain(
+    python_version = "3.11.5",
+)
+
+bazel_dep(name = "rules_pycross", version = "0.0.0")
+local_path_override(
+    module_name = "rules_pycross",
+    path = "../..",
+)
+
+environments = use_extension("@rules_pycross//pycross/extensions:environments.bzl", "environments")
+use_repo(environments, "pycross_environments")
+
+lock_import = use_extension("@rules_pycross//pycross/extensions:lock_import.bzl", "lock_import")
+lock_import.import_pdm(
+    all_development_groups = True,
+    lock_file = "//:pdm.lock",
+    project_file = "//:pyproject.toml",
+    repo = "pdm_deps",
+)
+
+lock_repos = use_extension("@rules_pycross//pycross/extensions:lock_repos.bzl", "lock_repos")
+use_repo(lock_repos, "pdm_deps")
+
+lock_file = use_extension("@rules_pycross//pycross/extensions:lock_file.bzl", "lock_file")
+lock_file.instantiate(
+    name = "pdm_lock_file",
+    lock_file = "//:pdm_lock.bzl",
+)
+use_repo(lock_file, "pdm_lock_file")
```

Comment:
I found the static lockfile very useful in debugging. And also, I intend to use such a setup in my own monorepo (getting as much as possible out of the analysis phase is a major draw to `rules_pycross`), so I wanted to ensure that monorepo-support works in that context.

Neither of these are particularly good reasons to keep it in the example though. Removed in the interest of decluttering.

Reply:
Good to know about your preference for the static file. The combination of targets and repositories in one file works with WORKSPACE, but is cumbersome with bzlmod to say the least. In bzlmod, the only place that repos can be created is within a module extension. And unlike WORKSPACE, that extension can't `load` the repo definitions from a file with a user-provided name. You can see [here](https://github.com/jvolkman/rules_pycross/blob/main/pycross/private/tools/extract_lock_repos.py) what I've resorted to for now to support those static files in bzlmod.

I've considered instead making the static file some other format - maybe TOML to fit in with the modern ecosystem. Would that work for you?

Reply:
Yeah, the lockfile has become a bit clunky under Bzlmod to be sure, but I think it may still be my best option. I don't really care what format is used for the lockfile. My main interest is in avoiding the need to rely on Python to work out what repos exist: system Python is too unreliable on dev laptops (esp. those of Python devs, who will be tweaking things), serializing execution behind a fresh download of cpython has been pretty bad for cold-start performance.